cloud-core/src/main/java/cloud/commandframework/CommandTree.java

```diff
@@ -629,15 +629,11 @@ private CommandTree(final @NonNull CommandManager<C> commandManager) {
 
         if (commandQueue.isEmpty()) {
             return Collections.emptyList();
-        } else if (child.isLeaf() && commandQueue.size() < 2) {
-            return this.directSuggestions(commandContext, child, commandQueue.peek());
         } else if (child.isLeaf()) {
-            if (child.getValue() instanceof CompoundArgument) {
-                final String last = ((LinkedList<String>) commandQueue).getLast();
-                commandContext.setCurrentArgument(child.getValue());
-                return child.getValue().getSuggestionsProvider().apply(commandContext, last);
-            }
-            return Collections.emptyList();
+            String input = commandQueue.size() == 1 ? commandQueue.peek()
```

Comment:
please make this final, and imo the double ternary is still readable but it might be better to have one of them be an if to make things more obvious